kotlinx-coroutines-core/native/src/Builders.kt

```diff
@@ -50,31 +56,47 @@ public fun <T> runBlocking(context: CoroutineContext = EmptyCoroutineContext, bl
     return coroutine.joinBlocking()
 }
 
+@SharedImmutable
+internal val EMPTY_BLOCK: () -> Unit = {}
```

Comment:
This particular case wasn't optimized by K/N for some reason. Performance impact is likely to be negligible, I've extracted it during performance investigation

Reply:
Do you mean, unlikely to be negligible? If so, could you drop a comment then that states that this is a performance optimization (not a workaround for a bug, not a functionally significant decision, etc.)?

Reply:
No, I've meant exactly that -- it's unlikely that this change affects anything. I've reverted it

Reply:
Ah, ok.
> I've extracted it during performance investigation

This left the impression that you extracted it because of the results of the investigation, so I got confused. Got it.